service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -164,6 +164,11 @@ public static DigitalTwinAsyncClient createFromConnectionString(String connectio
      * @throws IOException can be thrown if the provided payload cannot be deserialized into a valid Json object.
      */
     public Observable<DigitalTwinCommandResponse> invokeCommand (String digitalTwinId, String commandName, String payload) throws IOException {
+        // Retrofit does not work well with null in body
+        if(payload == null)
```

Comment:
You won't need this since it is covered in withResponse overload

Reply:
yes i've cleaned up all of this in the follow up PR.